### ISSUE TITLE
Non-global alternative to `--modules`

### ISSUE THREAD

**Issue by 6b6279** [paulgazz/kmax#230]
Hello,

I was wondering if there is a way to pre-set a tristate symbol to `m` while calling `klocalizer` without the `--modules` flag. Not all tristate symbols I want to pre-set have the value `m`, so using `--modules` isn’t an option. I was expecting something along the lines of the `-U` or the `-D` option.

Thanks in advance.

**Comment by paulgazz** [comment]
I don't think we this capability through the interface, but it should be possible to add via [get_config_from_model](https://github.com/paulgazz/kmax/blob/61461c210785e38e64550fb763662a36de3d1482/kmax/klocalizer.py#L399).  Could you say a little more about what behavior you are looking for?  `-D` and `-U` tell klocalizer to force the configuration option on or off (respectively).  Is this the behavior you'd like (but with the option set to `m`, e.g., with a `-M` flag) or would you just like the option set to `m` only when `klocalizer` determines the option to be enabled (i.e., it is either `m` or `n`)?

**Comment by 6b6279** [comment]
I am using `klocalizer` to test the validity of configuration files corresponding to an arbitrary Kconfig file (i.e. the Kconfig files aren’t necessarily from the Linux kernel source). I feed `klocalizer` with the output of `kextract` and `kclause` on a given Kconfig file and then enter the configuration to test over the `-D` and `-U` flags. There are of course Kconfig files with multiple tristate symbols and in that case there are configurations where one of the tristate symbols is set to `m`, and the other possibly to `t`. In that case I can’t give `klocalizer` the configuration using the `--modules` flag, because not all tristate symbols are set to `m`. So I am quite precisely looking for an `-M` option as you have described it in your comment.

**Comment by 6b6279** [comment]
Additionally, I have found a bug (?):

A dependency between config symbols is defined as a upper-bound relationship in the Linux kernel documentation. As in, if A depends on B, A can at most assume the value of B; alas A < B. In case of tristate symbols, if a boolean symbol depends on a tristate symbol, the boolean symbol cannot assume the value `t` if the tristate symbol is set to `m`, as `m` is lesser than `t` (there is a table in https://www.kernel.org/doc/html/next/kbuild/kconfig-language.html#menu-attributes). `klocalizer` seems ignore this:
```
config SYMBOL_A
 bool "Config symbol A"
 depends on SYMBOL_B

config SYMBOL_B
 tristate "Config symbol B"
```
`klocalizer` called with `--modules -DCONFIG_SYMBOL_B -DCONFIG_SYMBOL_A` says that the constraints are satisfiable.
